src/haxelib/client/Main.hx

```diff
@@ -843,6 +846,39 @@ class Main {
 		Cli.print('Local repository deleted ($path)');
 	}
 
+	/** Outputs a lock file with all the installed libraries **/
+	function lock() {
+		if (RepoManager.getLocalPath(Sys.getCwd()) == null) {
+			Cli.printWarning("no local repository found, dumping lock from global");
+		}
+
+		final scope = getScope();
+		final libraryInfo = scope.getArrayOfLibraryInfo();
+
+		// sort projects alphabetically
+		libraryInfo.sort(function(a, b) return Reflect.compare(a.name.toLowerCase(), b.name.toLowerCase()));
+
+		for (library in libraryInfo) {
+			if (library.devPath != null) {
+				Cli.printError('Error: version of "${library.name}" is "dev". The lockfile will not work properly in other environments!');
+			}
+
+			final version = scope.getVersion(library.name);
+			var versionData: VersionData;
+			if (VcsID.isValid(version)) {
+				final vcsId = VcsID.ofString(version);
+				versionData = VcsInstall(
+					vcsId,
+					scope.repository.getVcsData(library.name, vcsId)
+				);
+			} else {
+				versionData = Haxelib(SemVer.ofString(version));
+			}
+			final versionStr = VersionDataHelper.toString(versionData);
+			Cli.print('-L ${library.name}:$versionStr');
```

Comment:
I'm still not sure about generating `-L` instead of `--lib` because there are still people dealing with older versions. Might be just me, though..

Reply:
People using `haxelib lock` won't be using versions not supporting `-L` :)
I will change it to `--lib` anyway for clarity.

Reply:
> People using haxelib lock won't be using versions not supporting -L :)
I disagree. I would use haxelib lock but I also support CI for multiple haxe versions.